dotnet-client-libraries/ParsingHtmlWithAngleSharp/Tests/ParsingHtmlWithAngleSharpUnitTests.cs

```diff
@@ -0,0 +1,296 @@
+
+        Assert.Equal(userNameInputElement, userNameInputElementLinq);
+    }
+
+    [Fact]
+    public async Task WhenParsingHtmlFromString_ThenDocumentIsCreatedAndElementPropertiesAreAsExpected()
+    {
+        var config = Configuration.Default;
+        var context = BrowsingContext.New(config);
+
+        var document = await context.OpenAsync(req => req.Content(Html));
+
+        var section = document.Body!.QuerySelector<IHtmlElement>("section")!;
+
+        var sectionInnerHtml = section.InnerHtml;
+        var sectionTextContent = section.TextContent;
+        var sectionAttributes = section.Attributes;
+        var sectionChildren = section.Children; // IHtmlCollection<IElement>
+
+        var nextSibling = section.NextElementSibling;
+        var previousSibling = section.PreviousElementSibling;
+
+        Assert.Equal(@"
+                    <div id=""articles"">
+                                <article id=""a1"">Article 1 <em>content</em>.</article>
+                            </div>
+                            <p class=""paragraph"">This is a paragraph.</p>
+                            <ul id=""list"">
+                                <li class=""blue"">Item 1</li>
+                                <li>Item 2</li>
+                                <li class=""blue"">Item 3</li>
+                            </ul>
+                            <form id=""sign-up-form"">
+                                <label for=""username"">Username: </label>
+                                <input id=""username"" name=""username"" type=""text"">
+
+                                <label for=""password"">Password: </label>
+                                <input id=""password"" name=""password"" type=""password"">
+
+                                <button id=""button"" type=""submit"">Sign up</button>
+                            </form>
+                    ".Trim().Replace(Environment.NewLine, "\n"),
+            sectionInnerHtml.Trim());
+
+
+        Assert.Equal(1, sectionAttributes.Length);
+        Assert.Equal(4, sectionChildren.Length);
+
+        Assert.Equal("footer", nextSibling!.TagName.ToLower());
+        Assert.Equal("h2", previousSibling!.TagName.ToLower());
+    }
+
+    [Fact]
+    public async Task WhenParsingHtmlFromUrl_ThenDocumentIsCreatedAndBooksDataIsProperlyRetrieved()
+    {
+        var booksCatalogUrl = "https://books.toscrape.com/";
+
+        var config = Configuration.Default
+            .WithDefaultLoader()
+            .WithJs();
+        var context = BrowsingContext.New(config);
+
+        var document = await context.OpenAsync(new Url(booksCatalogUrl));
+
+        var booksSection = document.QuerySelector<IHtmlElement>("div.page_inner section")!;
+        var bookInfoArticles = booksSection
+            .QuerySelectorAll<IHtmlElement>("li > article.product_pod")
+            .ToCollection();
+
+        var books = bookInfoArticles.Select(ToBook).ToList();
+        Console.WriteLine(books.Count()); // 20
```

Comment:
@vnikolaew 
here is the instance I mentioned.

Reply:
@elfico Okay, shall I remove this section then?

Reply:
@vnikolaew ,
Yes, I think we should remove this ` Console.WriteLine(books.Count()); // 20` as it is redundant in this case. The others look ok

Reply:
@elfico I suppose I can remove it from one of the article's code snippets as well, right?